src/flutter/shell/platform/linux_embedded/window/native_window_drm_eglstream.cc

```diff
@@ -0,0 +1,299 @@
+                           drm_mode_info_.vdisplay << 16);
+  drmModeAtomicAddProperty(atomic, drm_plane_id_, property_ids.plane.crtc_x, 0);
+  drmModeAtomicAddProperty(atomic, drm_plane_id_, property_ids.plane.crtc_y, 0);
+  drmModeAtomicAddProperty(atomic, drm_plane_id_, property_ids.plane.crtc_w,
+                           drm_mode_info_.hdisplay);
+  drmModeAtomicAddProperty(atomic, drm_plane_id_, property_ids.plane.crtc_h,
+                           drm_mode_info_.vdisplay);
+  drmModeAtomicAddProperty(atomic, drm_plane_id_, property_ids.plane.fb_id,
+                           drm_fb_);
+  drmModeAtomicAddProperty(atomic, drm_plane_id_, property_ids.plane.crtc_id,
+                           drm_crtc_->crtc_id);
+  return true;
+}
+
+void NativeWindowDrmEglstream::GetPropertyIds(
+    struct NativeWindowDrmEglstream::drm_property_ids* property_ids) {
+  struct drm_property_address crtc_table[] = {
+      {"MODE_ID", &property_ids->crtc.mode_id},
+      {"ACTIVE", &property_ids->crtc.active},
+  };
+  struct drm_property_address plane_table[] = {
+      {"SRC_X", &property_ids->plane.src_x},
+      {"SRC_Y", &property_ids->plane.src_y},
+      {"SRC_W", &property_ids->plane.src_w},
+      {"SRC_H", &property_ids->plane.src_h},
+      {"CRTC_X", &property_ids->plane.crtc_x},
+      {"CRTC_Y", &property_ids->plane.crtc_y},
+      {"CRTC_W", &property_ids->plane.crtc_w},
+      {"CRTC_H", &property_ids->plane.crtc_h},
+      {"FB_ID", &property_ids->plane.fb_id},
+      {"CRTC_ID", &property_ids->plane.crtc_id},
+      {"CRTC_ID", &property_ids->connector.crtc_id},
+  };
+  struct drm_property_address connector_table[] = {
+      {"CRTC_ID", &property_ids->connector.crtc_id},
+  };
+
+  GetPropertyAddress(drm_crtc_->crtc_id, DRM_MODE_OBJECT_CRTC, crtc_table,
+                     sizeof(crtc_table) / sizeof(drm_property_address));
+  GetPropertyAddress(drm_plane_id_, DRM_MODE_OBJECT_PLANE, plane_table,
+                     sizeof(plane_table) / sizeof(drm_property_address));
+  GetPropertyAddress(drm_connector_id_, DRM_MODE_OBJECT_CONNECTOR,
+                     connector_table,
+                     sizeof(connector_table) / sizeof(drm_property_address));
+}
+
+void NativeWindowDrmEglstream::GetPropertyAddress(
+    uint32_t id, uint32_t type,
+    NativeWindowDrmEglstream::drm_property_address* table, size_t length) {
+  auto properties = drmModeObjectGetProperties(drm_device_, id, type);
+  if (properties) {
+    for (uint32_t i = 0; i < properties->count_props; i++) {
+      auto property = drmModeGetProperty(drm_device_, properties->props[i]);
+      if (property) {
+        for (uint32_t j = 0; j < length; j++) {
+          if (strcmp(table[j].name, property->name) == 0) {
+            *(table[j].ptr) = property->prop_id;
+            break;
+          }
+        }
+      }
+      drmModeFreeProperty(property);
+    }
+    drmModeFreeObjectProperties(properties);
+  }
+}
+
+bool NativeWindowDrmEglstream::CreatePropertyBlob() {
```

Comment:
Why did you create this function instead of using drmModeCreatePropertyBlob directly?

Reply:
I removed this function and fixed it to use `drmModeCreatePropertyBlob` directly.